### PR TITLE
Implementation of new embedding and fixes

### PR DESCRIPTION
**Pull request by gabrieledangeli**
Here's a brief description of my changes:

- `setup.py`: Fixed installation requirements (tested, should be good now).

- `embeggings.py`: Implemented the `PatchAmplitudeEmbedding` class. Added a ceiling function to ensure the correct number of address qubits in the `PatchEmbedding` class. Correct minor typos in the documentations and added a small offset to prevent NaN loss when using `PatchAmplitudeEmbedding`.

- `metrics.py`: Implemented the `CombinedOptaxLoss` class to handle combined Log Softmax Cross Entropy loss with L2 Regularization.

- `model.py`: Fixed a small bug in the `predict` function when checking the correct number of tensors in the MPS.

- `utils.py`: Fixed the `pad_images` function.

### CODE REVIEW

File: tn4ml/metrics.py
Comment (emapuljak) [inline tn4ml/metrics.py:1]: I will not approve the CombinedOptaxLoss because the all ready existing CombinedLoss could just be modified

File: tn4ml/models/model.py
Comment (emapuljak) [inline tn4ml/models/model.py:1]: Also this function is to compute accuracy, not return predictions. I will create a new function for this.

**Review by emapuljak — CHANGES_REQUESTED** [review]
- `CombinedOptaxLoss` should be combined with `CombinedLoss`
- Create function 'forward' in model.py that returns outputs of the model's forward pass, which equals predictions

File: tn4ml/models/model.py
Comment (emapuljak) [reply tn4ml/models/model.py:1]: Did it :)

**Review by emapuljak — COMMENTED** [review]
Its okay. But I will only merge additional functions

**Comment by emapuljak** [comment]
I will close this PR for now. Please create new one only requesting to implement the new embedding function.
